perllib/FixMyStreet/App/Controller/Admin/Bodies.pm

```diff
@@ -234,6 +234,14 @@ sub update_contact : Private {
 
     my $category = $self->trim( $c->get_param('category') );
     $errors{category} = _("Please choose a category") unless $category;
+
+    my $body = $c->model('DB::Body')->find({
+        id => $c->stash->{body_id},
+    });
+    my $cobrand = $body->get_cobrand_handler;
+    my $category_validation_error = $cobrand ? $cobrand->call_hook(validate_category => $category) : "";
```

Comment:
I think 'validate_category' is a bit generic for a cobrand hook (in that is it to do with public reporting, this admin, some Open311 sending?) - perhaps something like `admin_contact_validate_category`?